src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java

```diff
@@ -94,6 +94,27 @@ public SampleEndpoint(SampleService sampleService, MapperFacade mapperFacade) {
     return ResponseEntity.ok(result);
   }
 
+  /**
+   * DELETE endpoint to delete the sample summary and all related sample units given sampleSummaryId
+   *
+   * @param sampleSummaryId the id fo the SampleSummary to delete
+   * @return noContent on success
+   * @throws CTPException if SampleSummary not found
+   */
+  @RequestMapping(value = "/samplesummary/{sampleSummaryId}", method = RequestMethod.DELETE)
+  public ResponseEntity<?> deleteSampleSummary(
+      @PathVariable("sampleSummaryId") final UUID sampleSummaryId) throws CTPException {
+    SampleSummary sampleSummary = sampleService.findSampleSummary(sampleSummaryId);
+    if (sampleSummary == null) {
+      throw new CTPException(
+          CTPException.Fault.RESOURCE_NOT_FOUND,
+          String.format("Sample Summary not found for sampleSummaryId %s", sampleSummaryId));
+    }
+    sampleService.deleteSampleSummaryAndSampleUnits(sampleSummary);
```

Comment:
Just a question: how long will this process take if the sample load is in 1000s

Reply:
I realise I forgot to test this.  When I ran it via raw SQL commands in the papst, it was pretty fast.  I'm hoping it won't be much slower